tests/testthat/test-linter_tags.R

```diff
@@ -27,3 +27,69 @@ test_that("available_linters matches the set of linters available from lintr", {
   # ensure that all _linter objects in our namespace are also exported
   expect_setequal(all_linters, grep("_linter$", getNamespaceExports("lintr"), value = TRUE))
 })
+
+test_that("lintr help files are up to date", {
+  helper_db_dir <- system.file("help", package = "lintr")
+  skip_if_not(dir.exists(helper_db_dir))
+  skip_if_not(file.exists(file.path(helper_db_dir, "lintr.rdb")))
+  skip_if_not(file.exists(file.path(helper_db_dir, "lintr.rdx")))
+
+  help_env <- new.env(parent = topenv())
+  lazyLoad(file.path(helper_db_dir, "lintr"), help_env)
+
+  lintr_db <- available_linters()
+
+  # some of the following might seem redundant, so note: we want to establish
+  #   that entries exist in the help _if and only if_ they exist in the csv in inst/.
+  #   so e.g. checking that all linters in the DB match a line in the Rd is not enough,
+  #   e.g. if a deleted/renamed linter's entry still persists in the help page, and vice versa
+
+  expect_true(exists("linters", envir = help_env))
+  # objects in help_env are class Rd, see ?as.character.Rd (part of 'tools')
+  linter_help_text <- paste(as.character(help_env$linters), collapse = "")
+
+  # Rd markup for items looks like \item{\code{\link{...}} (tags: ...)}
+  linter_item_regex <- "[\\]item[{][\\]code[{][\\]link[{][a-zA-Z0-9._]+[}][}] [(]tags: [a-z_, ]+[)][}]"
+  expect_identical(
+    length(gregexpr(linter_item_regex, linter_help_text)[[1L]]),
+    nrow(lintr_db),
+    info = "Count of linter entries in ?linters matches inst/lintr/linters.csv row count"
+  )
```

Comment:
Both are sorted, so if we add a capture group for the linter function name, we could use `expect_equal()` which would provide us with the offending linters. WDYT?

Reply:
Seems a bit fragile to me since the sorting is coming from different origins.

How about capture group + `sort()` here?

Reply:
PTAL. hopefully easier to understand with the new approach. And moreover, it already worked -- it caught that the count for `consistency_linters` is out of date :)